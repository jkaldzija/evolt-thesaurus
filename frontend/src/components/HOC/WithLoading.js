import React from 'react';
import style from './loader.module.css';
import PropTypes from 'prop-types';
import {Spinner} from "reactstrap";

export function WithLoading(Component) {

    return class extends React.Component {

        constructor(props) {
            super(props);

            this.state = {
                hidden: true
            }
        }


        componentDidUpdate(prevProps, prevState) {
            if(this.props.isLoading != prevProps.isLoading){
                if(this.props.isLoading){
                    this.interval = setTimeout(() => {
                        this.setState({ hidden: false });
                    }, 200);
                }
                else{
                    clearInterval(this.interval);
                    this.setState({ hidden: true });
                }
            }
        }


        render(){
            const {isLoading} = this.props;
            if (isLoading && !this.state.hidden) {
                    return <div className="d-flex flex-column justify-content-center align-items-center">
                        <Spinner className={style.spinner} color={"#3cbd3f"} />
                        <div className="dark-gray-text">Loading, please wait</div>
                    </div>;
            }
            return (<Component {...this.props} />);
        }

    }
}
