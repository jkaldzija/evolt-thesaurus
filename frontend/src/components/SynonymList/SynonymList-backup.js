import React, {Fragment} from 'react';
import {Typography} from "@material-ui/core";
import type {Word} from "../../api/types";

type SynonymListProps = {
    word: Word
};

const renderSynonyms = (word: Word) => (
    <Fragment>
        <Typography variant="h6" className="synonym-list-title">Synonyms for {word.value}</Typography>
        {word.synonyms.map(synonym => (
            <Typography key={synonym} variant="body1">{synonym}</Typography>
        ))}
    </Fragment>
);

const renderEmpty = (word: Word) => (
    <div className="synonym-empty-container">
        <Typography variant="h5">No synonyms found for {word.value}</Typography>
    </div>
);

const SynonymList = ({word}: SynonymListProps) => {
    if (word.synonyms) {
        return renderSynonyms(word)
    } else {
        return renderEmpty(word);
    }
};

export default SynonymList;
