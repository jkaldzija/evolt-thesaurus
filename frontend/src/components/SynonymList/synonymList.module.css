.synonym-list-title {
    margin-bottom: 12px !important;
}

.synonym-empty-container {
    display: flex;
    align-items: center;
    justify-content: center;
}

.wrapper{
    margin-bottom: 15px;
}

.link{
    background-color: #88cbe4;
    padding: 5px 10px;
    border-radius: 10px;
    color: white;
    margin-right: 10px;
    font-family: "Museo Sans Rounded 500";
}



.link:hover{
    color: white;
    text-decoration: none;
}

.link-button{
    padding: 5px 10px;
}
