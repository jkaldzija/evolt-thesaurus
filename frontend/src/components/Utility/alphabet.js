export const alphabet = {
    lower: 'abcdefghijklmnopqrstuvwxyz'.split(''),
    upper: 'abcdefghijklmnopqrstuvwxyz'.toUpperCase().split('')
};
