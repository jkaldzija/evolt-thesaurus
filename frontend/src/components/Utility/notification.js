import {toast} from "react-toastify";

const settings = {
    position: 'bottom-right',
    autoClose: true,
    hideProgressBar: false,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true
};

export function notifyError(message){
    createMessage(message, 'error');
}

export function notifySuccess(message){
    createMessage(message, 'success');
}

export function createMessage(message, type = 'success'){

    switch(type){
        case 'error':
            toast.error(message, settings);
            break;
        default:
            toast.success(message, settings);
            break;
    }

}


