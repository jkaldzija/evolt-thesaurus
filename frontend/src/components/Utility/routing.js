import * as qs from "qs";

export const routes = {
    home:{
        path: '/'
    },
    dictionary:{
        path: '/dictionary'
    }
};

export function redirect(history, route){
    history.push(route);
}

export function getSynonymRoute(synonym){
    return `${routes.home.path}?q=${synonym}`;
}

export function getQueryParam(query, param){
    return qs.parse(query, { ignoreQueryPrefix: true })[param];
}
