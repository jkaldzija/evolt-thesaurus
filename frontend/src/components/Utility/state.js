export const getSafe = (obj, def = "") => obj !== null && obj !== undefined ? obj : def;


export function getSafeDeep(object, path, undefinedValue=""){

    let currentObject = object;
    const pathStrings = path.split(".");


    for(let pathString of pathStrings){
        if(currentObject == null || currentObject == undefined) return undefinedValue;
        currentObject = currentObject[pathString];
    }
    if(currentObject == null || currentObject == undefined) return undefinedValue;
    return currentObject;
}

export function eliminateWhiteSpace(string) {
    return string.trimStart().replace("  ", " ");
}
