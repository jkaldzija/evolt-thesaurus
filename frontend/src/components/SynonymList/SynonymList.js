import React, {Fragment} from 'react';
import classnames from "classnames";
import style from "./synonymList.module.css";
import {Link} from "react-router-dom";
import ThemeButton from "../Form/ThemeButton/ThemeButton";
import AddWordDialog from "../AddWord/AddWordDialog";
import {getSynonymRoute} from "../Utility/routing";

type SynonymListProps = {
    synonymList: (Array),
    reloadData: () => void
}


class SynonymList extends React.Component<SynonymListProps> {
    constructor(props) {
        super(props);

        this.state = {
            addSynonymDialogOpen: false
        }
    }

    toggleDialog = (key) => () => {
        const fullKey = `${key}DialogOpen`;
        this.setState({ [fullKey]: !this.state[fullKey] });
    };


    renderSynonymList(){
        const {synonymList} = this.props;
        if(!synonymList || synonymList.length == 0){
            return <div className="d-flex align-items-center mr-3">No synonyms found.</div>
        }
        return <Fragment>
            {synonymList.map((synonym, index) => {
                return <Link key={`${synonym}-${index}`} className={style.link} to={getSynonymRoute(synonym)}>{synonym}</Link>
            })}
        </Fragment>
    }

    render() {
        return <div>
            <div className={classnames("d-flex flex-wrap", style.wrapper)}>
                {this.renderSynonymList()}
                <ThemeButton onClick={this.toggleDialog("addSynonym")} className={style["link-button"]}>
                     Add Synonym
                </ThemeButton>
            </div>
            <AddWordDialog
                word={this.props.currentWord}
                reloadData={this.props.reloadData}
                isOpen={this.state.addSynonymDialogOpen}
                title={"Add synonym"}
                toggle={this.toggleDialog("addSynonym")}
            />
        </div>
    }
}

// Set default props
SynonymList.defaultProps = {
    synonymList:[],
    reloadData: () => {}
};



export default SynonymList;
