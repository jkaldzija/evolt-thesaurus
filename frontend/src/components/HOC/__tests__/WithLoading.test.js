import React from 'react';
import {shallow} from "enzyme";
import {WithLoading} from "../WithLoading";
import {getSafeDeep} from "../../Utility/state";
import SynonymList from "../../SynonymList/SynonymList";

describe('With Loading', function () {
    it('should render correctly', function () {
        const SynonymListWithLoading = WithLoading(SynonymList);
        const component = shallow(
            <SynonymListWithLoading
                isLoading={true}  />
        ).dive();
        expect(component).toMatchSnapshot();
    });

    it('should show spinner', async function() {
        const SynonymListWithLoading = WithLoading(SynonymList);
        const component = shallow(
            <SynonymListWithLoading
                isLoading={false}  />
        );
        component.setProps({isLoading: true});

        await new Promise(resolve => setTimeout(resolve, 210));

            expect(component.state("hidden")).toEqual(false);
    });

    it('should hide spinner', async function() {
        const SynonymListWithLoading = WithLoading(SynonymList);
        const component = shallow(
            <SynonymListWithLoading
                isLoading={true}  />
        );
        component.setProps({isLoading: false});


        expect(component.state("hidden")).toEqual(true);
    });
});
