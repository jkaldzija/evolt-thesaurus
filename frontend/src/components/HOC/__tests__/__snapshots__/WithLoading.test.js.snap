// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Loading should render correctly 1`] = `
<div>
  <div
    className="d-flex flex-wrap wrapper"
  >
    <div
      className="d-flex align-items-center mr-3"
    >
      No synonyms found.
    </div>
    <ThemeButton
      className="link-button"
      onClick={[Function]}
    >
      Add Synonym
    </ThemeButton>
  </div>
  <AddWordDialog
    isOpen={false}
    open={false}
    reloadData={[Function]}
    title="Add synonym"
    toggle={[Function]}
    word={
      Object {
        "value": "",
      }
    }
  />
</div>
`;
